### DISCUSSION TITLE
Why can't ReplaceNode take a collection of nodes?

### DISCUSSION THREAD

**Discussion by bradwilson** [dotnet/roslyn#70897] [General]
I have a fix operation which replaces a single statement with multiple statements. I would like to use [`SyntaxEditor.ReplaceNode()`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.editing.syntaxeditor.replacenode), but it can only replace a single node with a single node, so I end up writing [insert then remove operations](https://github.com/xunit/xunit.analyzers/blob/e768a8ba50a8b1dd9ed2ab7c83ae24b7750dc107/src/xunit.analyzers.fixes/X2000/AssertSingleShouldBeUsedForSingleParameterFixer.cs#L165-L166).

I understand this is only saving me one line of code 😄 but I'm curious why there is no overload for `ReplaceNode` that allows you to provide multiple `SyntaxNode`s.

**Comment by huoyaoyuan** [comment]
How to ensure the relationship between these nodes? What happens if they are not continuous elements?

**Reply by bradwilson** [reply]
I am able to do the operation with a single insert, so it should do the exact same as that.

**Comment by CyrusNajmabadi** [comment]
Because no one created such an API. We would need such an idea to go through the standard api proposal route.   Thanks.
